kernel/dynamic.c

```diff
@@ -27,6 +27,9 @@ struct dyn_cb_data {
 	k_thread_stack_t *stack;
 };
 
+BUILD_ASSERT(K_THREAD_STACK_LEN(CONFIG_DYNAMIC_THREAD_POOL_SIZE) >= \
+	     K_KERNEL_STACK_LEN(CONFIG_DYNAMIC_THREAD_POOL_SIZE));
```

Comment:
This is definitely incorrect.

I think the assertion aims to say that thread stack sizes are greater than or equal to kernel stack sizes, but I don't believe we need that assertion here.

```suggestion
BUILD_ASSERT(K_THREAD_STACK_LEN(CONFIG_DYNAMIC_THREAD_STACK_SIZE) >= \
	     K_KERNEL_STACK_LEN(CONFIG_DYNAMIC_THREAD_STACK_SIZE));
```